src/devices/mindy.ts

```diff
@@ -0,0 +1,366 @@
+
+const e = exposes.presets;
+const ea = exposes.access;
+
+const tzLocal = {
+    last_boot: {
+        key: ["last_boot"],
+        convertGet: async (entity, key, meta) => {
+            await entity.read("genTime", ["lastSetTime"]);
+        },
+    } satisfies Tz.Converter,
+    wifi: {
+        key: ["wifi"],
+        convertGet: async (entity, key, meta) => {
+            await entity.read("genBasic", ["locationDesc"]);
+        },
+        convertSet: async (entity, key, value, meta) => {
+            await entity.write("genBasic", {locationDesc: value === "ON" ? "1" : "0"});
+
+            const retryRead = async (attempts: number) => {
+                if (attempts > 0) {
+                    try {
+                        await entity.read("genBasic", ["locationDesc"]);
+                    } catch (error) {
+                        /* Do nothing */
+                    }
+                    setTimeout(() => retryRead(attempts - 1), 10000);
+                }
+            };
+
+            retryRead(3);
+        },
+    } satisfies Tz.Converter,
+};
+
+const fzLocal = {
+    last_boot: {
+        cluster: "genTime",
+        type: ["readResponse"],
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.lastSetTime) {
+                const data = msg.data.lastSetTime;
+
+                // Assuming the epoch is January 1, 2000
+                const epoch = new Date("2000-01-01T00:00:00Z").getTime() / 1000;
+                const date = new Date((epoch + data) * 1000); // Convert seconds to milliseconds
+                const boot_date_time = date.toISOString();
+
+                return {last_boot: boot_date_time};
+            }
+        },
+    } satisfies Fz.Converter,
+    wifi: {
+        cluster: "genBasic",
+        type: ["readResponse"],
+        convert: (model, msg, publish, options, meta) => {
+            if (msg.data.locationDesc) {
+                const data = msg.data;
+                const wifi = data.locationDesc && data.locationDesc.trim().length > 0 ? "ON" : "OFF";
+
+                return {ip_address: data.locationDesc.trim(), wifi: wifi};
+            }
+        },
+    } satisfies Fz.Converter,
+};
+
+const optionsLocal = {
+    last_boot_update: () => {
+        return e
+            .numeric("last_boot_update", ea.STATE_SET)
+            .withValueMin(10)
+            .withDescription("Interval for request boot datetime from device. (default 60 seconds)");
+    },
+};
+
+export const definitions: DefinitionWithExtend[] = [
+    {
+        zigbeeModel: ["Leleka"],
+        model: "Leleka",
+        vendor: "MindY",
+        description: "Advanced Environmental Monitoring Device",
+        ota: true,
+        fromZigbee: [fzLocal.last_boot, fzLocal.wifi],
+        toZigbee: [tzLocal.last_boot, tzLocal.wifi],
+        options: [optionsLocal.last_boot_update()],
+        onEvent: (type, data, device, options) => {
+            const endpoint = device.getEndpoint(1);
```

Comment:
```suggestion
            const poll = async () => {
                        await endpoint.read("genTime", ["lastSetTime"]);
                        await endpoint.read("genBasic", ["locationDesc"]);
            };

            utils.onEventPoll(type, data, device, options, "last_boot_update", 60, poll);
```

You can use this instead

Reply:
I thought I understood your idea correctly.